### PR TITLE
Demo application and fixes

### PR DESCRIPTION
**Pull request by nitzann**

### CODE REVIEW

File: app/src/main/java/com/ibm/mobilefirstplatform/appid/MainActivity.java [hunk 1]
```diff
@@ -234,41 +257,41 @@ public void run () {
...
-                        public void run() {
-                            ImageView profilePicture = (ImageView) findViewById(R.id.profilePic);
+					//run on main thread
+					runOnUiThread(new Runnable() {
+						@Override
+						public void run() {
+							ImageView profilePicture = (ImageView) findViewById(R.id.profilePic);
 							if(bmp == null) {
-								profilePicture.setImageResource(R.mipmap.ic_launcher);
+								profilePicture.setImageResource(R.drawable.ic_anon_user);
```
Comment (rotembr) [inline app/src/main/java/com/ibm/mobilefirstplatform/appid/MainActivity.java:274]: From where this is taken, I get error here

File: lib/src/main/java/com/ibm/bluemix/appid/android/api/LoginWidget.java [hunk 2]
```diff
@@ -17,8 +17,13 @@
...
 public interface LoginWidget {
-	// TODO: What is the usecase for this method?
-//	void launch (@NonNull final Activity activity, @NonNull final AuthorizationListener authorizationListener, String accessTokenString);
+	/**
+	 * Lunch the login widget user interface
+	 * @param activity Parent activity
+	 * @param authorizationListener
+	 * @param accessTokenString
+	 */
+	void launch (@NonNull final Activity activity, @NonNull final AuthorizationListener authorizationListener, String accessTokenString);
```
Comment (rotembr) [inline lib/src/main/java/com/ibm/bluemix/appid/android/api/LoginWidget.java:26]: why we need this API?

Comment (nitzann) [reply lib/src/main/java/com/ibm/bluemix/appid/android/api/LoginWidget.java:26]: 1. This is a behavior that is supported by the REST API and also on the Swift SDK, so we need to be consistent and support it on Android
2. There is no reason to not allow this, its a capability we get for free
3. Use cases:
a. an application flow where the developer allows multiple anonymous logins and then wants to choose the one he wants to upgrade to identified, not necessarily the last one.
b. the developer wants to store away the anonymous token and use it in the upgrade if the application got restarted before the user logged in with identity.

File: app/src/main/java/com/ibm/mobilefirstplatform/appid/MainActivity.java (hunk 1 above)
Comment (nitzann) [reply app/src/main/java/com/ibm/mobilefirstplatform/appid/MainActivity.java:274]: uploaded the missing images

**Review by rotembr — APPROVED** [review]

**Pull request merged**